llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -2357,41 +2323,43 @@ bool SIInsertWaitcnts::shouldFlushVmCnt(MachineLoop *ML,
         if (MI.mayStore())
           HasVMemStore = true;
       }
-      for (unsigned I = 0; I < MI.getNumOperands(); I++) {
-        MachineOperand &Op = MI.getOperand(I);
+      for (const MachineOperand &Op : MI.uses()) {
```

Comment:
Looks like the old code also handled implicit uses, so this should be `MI.all_uses()`?

Reply:
Done